pkg/core/blockchain.go

```diff
@@ -1112,48 +1112,10 @@ func (bc *Blockchain) references(ins []transaction.Input) ([]transaction.InOut,
 	return references, nil
 }
 
-// FeePerByte returns network fee divided by the size of the transaction.
-func (bc *Blockchain) FeePerByte(t *transaction.Transaction) util.Fixed8 {
-	return bc.NetworkFee(t).Div(int64(io.GetVarSize(t)))
-}
-
-// NetworkFee returns network fee.
-func (bc *Blockchain) NetworkFee(t *transaction.Transaction) util.Fixed8 {
-	// https://github.com/neo-project/neo/blob/master-2.x/neo/Network/P2P/Payloads/ClaimTransaction.cs#L16
-	if t.Type == transaction.ClaimType {
-		return 0
-	}
-
-	inputAmount := util.Fixed8FromInt64(0)
-	refs, err := bc.References(t)
-	if err != nil {
-		return inputAmount
-	}
-	for i := range refs {
-		if refs[i].Out.AssetID == UtilityTokenID() {
-			inputAmount = inputAmount.Add(refs[i].Out.Amount)
-		}
-	}
-
-	outputAmount := util.Fixed8FromInt64(0)
-	for _, txOutput := range t.Outputs {
-		if txOutput.AssetID == UtilityTokenID() {
-			outputAmount = outputAmount.Add(txOutput.Amount)
-		}
-	}
-
-	return inputAmount.Sub(outputAmount).Sub(bc.SystemFee(t))
-}
-
-// SystemFee returns system fee.
-func (bc *Blockchain) SystemFee(t *transaction.Transaction) util.Fixed8 {
-	if t.Type == transaction.InvocationType {
-		inv := t.Data.(*transaction.InvocationTX)
-		if inv.Version >= 1 {
-			return inv.Gas
-		}
-	}
-	return bc.GetConfig().SystemFee.TryGetValue(t.Type)
+// FeePerByte returns transaction network fee per byte.
+// TODO: should be implemented as part of PolicyContract
+func (bc *Blockchain) FeePerByte(_ *transaction.Transaction) util.Fixed8 {
```

Comment:
So why `*transaction.Transaction` parameter if it doesn't depend on it? IIUC it won't depend on it even with native policy contract, it was old Neo 2 where this dependency existed.

Reply:
Removed the parameter, now it's just a `FeePerByte()`.